### PR TITLE
Save audit logs command

### PR DESCRIPTION
**Pull request by notevanrutten**
Everything works fine, but for some reason it throws an error when run through the slash command (which doesn't affect functionality). The error states an "unknown interaction", as if there is somehow a second interaction after the command is run. The issue doesn't impact the bot in any way other than cluttering the console logs.

### CODE REVIEW

File: src/commands/save-audit-logs.mts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            await interaction.reply("You don't have the required role to use this command.");
+            return;
+        }
+
+        // Call the function to save the audit logs, with interaction provided to indicate the manual scenario
+        await saveAuditLogs(interaction, guild);
+    },
+} as Command;
+
+export const saveAuditLogs = async (interaction: CommandInteraction | null, guild: Guild) => {
```
Comment (Nomalah) [inline src/commands/save-audit-logs.mts:33]:
```suggestion
export async function saveAuditLogs(interaction: CommandInteraction | null, guild: Guild) {
```
Prefer functions over lambdas if they're named

File: src/commands/save-audit-logs.mts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+// Logs are saved to a file locally which is also posted to the channel
+// Will run weekly, but can still be invoked manually by the user
+
+import { CommandInteraction, Guild, SlashCommandBuilder, TextChannel } from "discord.js";
+import { Command } from "../types.mjs";
+import fs from "fs";
+import path from "path";
+
+export default {
+    data: new SlashCommandBuilder().setName("save-audit-logs").setDescription("Save audit logs from the past week."),
```
Comment (Nomalah) [inline src/commands/save-audit-logs.mts:11]: Could you add a date parameter on when save audit logs since?

File: src/events/ready.mts [hunk 3]
```diff
@@ -7,9 +12,36 @@ export default {
...
     execute(client: Client) {
-        if (client.user === null) {
+        if (!client.user) {
             throw new Error("client.user is null");
         }
         console.log(`Ready! Logged in as ${client.user.tag}`);
+
+        // Schedule weekly audit log saving:
+        // Saturday @ 11:59 PM
+        cron.schedule("59 23 * * 6", async () => {
```
Comment (Nomalah) [inline src/events/ready.mts:22]: Just do Sunday @ Midnight
```suggestion
        cron.schedule("0 0 * * 0", async () => {
```

File: src/events/ready.mts [hunk 4]
```diff
@@ -7,9 +12,36 @@ export default {
...
+            const guild_id = process.env.DISCORD_GUILD_ID;
+
+            // Check if guild_id is undefined
+            if (!guild_id) {
+                console.error("DISCORD_GUILD_ID environment variable is not set.");
+                return;
+            }
```
Comment (Nomalah) [inline src/events/ready.mts:29]: Assume it's set, or else the bot won't even log in.
```suggestion
```

**Review by Nomalah — CHANGES_REQUESTED** [review]
Looks pretty good, just a few things to clean up + add, and we'll be set.

File: src/events/ready.mts (hunk 3 above)
Comment (notevanrutten) [reply src/events/ready.mts:22]: I can do this, just gotta change the dates in the file name to start at the previous Sunday and end at the Saturday right before the logs are saved. That's if we ignore the fact that it technically goes one minute over the time period.

File: src/events/ready.mts (hunk 4 above)
Comment (notevanrutten) [reply src/events/ready.mts:29]: Resolved in a separate commit:
Added `!` after both instances of `guild_id`

File: src/commands/save-audit-logs.mts (hunk 2 above)
Comment (notevanrutten) [reply src/commands/save-audit-logs.mts]: Taken care of in another commit

File: src/events/ready.mts (hunk 3 above)
Comment (notevanrutten) [reply src/events/ready.mts:22]: Going to keep the end time at 11:59 Saturday since it makes more sense for the dates.

**Review by Nomalah — APPROVED** [review]

**Pull request merged**